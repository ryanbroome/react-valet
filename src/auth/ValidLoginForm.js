import React, { useContext } from "react";
import { useFormik } from "formik";
import * as Yup from "yup";

import { FormGroup, Form, Input, Label, Button } from "reactstrap";
import UserContext from "./UserContext";

const ValidLoginForm = () => {
  const { login, error, setError } = useContext(UserContext);

  const formik = useFormik({
    initialValues: {
      username: "",
      password: "",
    },
    validationSchema: Yup.object({
      username: Yup.string().max(25, "Must be 25 characters or less").required("username required"),
      password: Yup.string().required("password required"),
    }),
    onSubmit: async (values) => {
      login(values.username, values.password);
    },
  });

  return (
    <Form onSubmit={formik.handleSubmit}>
      <FormGroup>
        <Label htmlFor="username">username</Label>
        <Input
          id="username"
          placeholder="enter username here"
          name="username"
          type="text"
          autoComplete="username"
          onChange={formik.handleChange}
          onBlur={formik.handleBlur}
          value={formik.values.username}></Input>
        {formik.touched.username && formik.errors.username ? <div style={{ color: "red" }}>{formik.errors.username}</div> : null}

        <Label htmlFor="password">password</Label>
        <Input
          id="password"
          placeholder="enter password here"
          name="password"
          type="password"
          autoComplete="current-password"
          onChange={formik.handleChange}
          onBlur={formik.handleBlur}
          value={formik.values.password}></Input>
        {formik.touched.password && formik.errors.password ? <div style={{ color: "red" }}>{formik.errors.password}</div> : null}
      </FormGroup>
      <Button
        type="submit"
        color="success">
        Login
      </Button>
    </Form>
  );
};

export default ValidLoginForm;
