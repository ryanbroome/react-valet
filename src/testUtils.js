import React from "react";
import UserContext from "./auth/UserContext";
// import { render } from "@testing-library/react";
// import { MemoryRouter } from "react-router";
import App from "../src/App";
// import TransactionList from "./TransactionList";

const MockUserContext = ({ children }) => {
  const mockToken = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VybmFtZSI6InRlc3R1c2VyIiwibG9jYXRpb25JZCI6MSwiaXNBZG1pbiI6ZmFsc2UsImlhdCI6MTcxMzA3NTM4Mn0.dXlnawfzk6wlsgpouoSp58QLvlOOSJFC03iq_FXab-I";
  const mockUserDetail = {
    userId: 1,
    vehicleId: 1,
    locationId: 1,
    transactionId: 1,
    ticketNum: "1",
    mobile: "5553334444",
    color: "red",
    make: "honda",
    username: "testuser",
  };
  const mockLocation = {};
  const mockVehicle = {};
  const mockTransaction = {};

  return <UserContext.Provider value={{ token: mockToken, userDetail: mockUserDetail, location: mockLocation, vehicle: mockVehicle, transaction: mockTransaction, update: App.update }}>{children}</UserContext.Provider>;
};

export { MockUserContext };
