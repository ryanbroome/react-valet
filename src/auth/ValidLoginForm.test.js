import React from "react";
import { render } from "@testing-library/react";
import ValidLoginForm from "./ValidLoginForm";
import { MemoryRouter } from "react-router";
import { MockUserContext } from "../testUtils";

it("ValidLoginForm: matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <MockUserContext children={<ValidLoginForm />}></MockUserContext>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
