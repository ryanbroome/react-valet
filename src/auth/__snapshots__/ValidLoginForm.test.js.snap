// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ValidLoginForm: matches snapshot 1`] = `
<DocumentFragment>
  <form
    class=""
  >
    <div
      class="position-relative form-group"
    >
      <label
        class=""
        for="username"
      >
        username
      </label>
      <input
        autocomplete="username"
        class="form-control"
        id="username"
        name="username"
        placeholder="enter username here"
        type="text"
        value=""
      />
      <label
        class=""
        for="password"
      >
        password
      </label>
      <input
        autocomplete="current-password"
        class="form-control"
        id="password"
        name="password"
        placeholder="enter password here"
        type="password"
        value=""
      />
    </div>
    <button
      class="btn btn-success"
      type="submit"
    >
      Login
    </button>
  </form>
</DocumentFragment>
`;
